mathics/core/definitions.py

```diff
@@ -480,76 +532,111 @@ def get_definition(self, name: str, only_if_exists=False) -> Optional["Definitio
         result.sort()
         return result
 
-    def get_nvalues(self, name):
+    def get_nvalues(self, name: str):
         return self.get_definition(name).nvalues
 
-    def get_defaultvalues(self, name):
+    def get_defaultvalues(self, name: str):
         return self.get_definition(name).defaultvalues
 
-    def get_value(self, name, pos, pattern, evaluation):
+    def get_value(self, name: str, pos: str, pattern, evaluation):
         assert isinstance(name, str)
         assert "`" in name
         rules = self.get_definition(name).get_values_list(valuesname(pos))
         for rule in rules:
             result = rule.apply(pattern, evaluation)
             if result is not None:
                 return result
+        return None
+
+    def get_user_definition(self, name: str, create: bool = True) -> "Definition":
+        """
+        Return a user definition for `name`. If `create` is `False`
+        and a user definition is not available, raise a KeyError exception.
+
+        Otherwise, tries to create a definition from a definition existing
+        into the space of builtin definitions.
+
+        Parameters
+        ----------
+        name : str
+            The name of the requested symbol.
+        create : bool, optional
+            If `True` and the symbol does not have a definition, create it.
+            Otherwise, raise a KeyError exception. The default is True.
+
+        Raises
+        ------
+        KeyError
+
+        Returns
+        -------
+        Optional[Definition]
+            The definition of the symbol.
+
+        """
 
-    def get_user_definition(self, name, create=True) -> Optional["Definition"]:
         assert not isinstance(name, Symbol)
 
         existing = self.user.get(name)
         if existing:
             return existing
+
+        if not create:
+            raise KeyError(name)
+        builtin = self.builtin.get(name)
+        if builtin:
+            attributes = builtin.attributes
+            is_numeric = builtin.is_numeric
         else:
-            if not create:
-                return None
-            builtin = self.builtin.get(name)
-            if builtin:
-                attributes = builtin.attributes
-                is_numeric = builtin.is_numeric
-            else:
-                attributes = A_NO_ATTRIBUTES
-                is_numeric = False
-            self.user[name] = Definition(
-                name=name,
-                attributes=attributes,
-                is_numeric=is_numeric,
-            )
-            self.clear_cache(name)
-            return self.user[name]
+            attributes = A_NO_ATTRIBUTES
+            is_numeric = False
+        self.user[name] = Definition(
+            name=name,
+            attributes=attributes,
+            is_numeric=is_numeric,
+        )
+        self.clear_cache(name)
+        return self.user[name]
 
     def mark_changed(self, definition: "Definition") -> None:
```

Comment:
Same here with respect to removing quotes around "Definition". And it also suggests why the Definition class should be defined first.

Reply:
Done.